File: backend/src/main/scala/model/Asset.scala
package model

import zio.json._

case class Asset(
                  id: Int = 0,
                  name: String,
                  host: String,
                  entityType: String,
                  username: String,
                  password: String
                )

object Asset {
  implicit val decoder: JsonDecoder[Asset] =
    DeriveJsonDecoder.gen[Asset]

  implicit val encoder: JsonEncoder[Asset] =
    DeriveJsonEncoder.gen[Asset]
}

File: frontend/src/types/Asset.ts
export type Asset = {
    id: number;
    name: string;
    host: string;
    entityType: string;
    username: string;
    password: string;
};

export type AssetResponse = {
    id: number;
    name: string;
    host: string;
    entityType: string;
    username: string;
};

export type AssetCreateRequest = {
    name: string;
    host: string;
    entityType: string;
    username: string;
    password: string;
    action?: "new" | "overwrite";
    overwriteId?: number;
};
